stage0/src/sev.rs

```diff
@@ -513,8 +513,17 @@ pub fn validate_memory(e820_table: &[BootE820Entry], encrypted: u64) {
     // the RMP for the fw_cfg DMA buffer.
     let min_addr = 0xA0000;
 
+    // Locate the legacy SMBIOS range in the E820 table and verify it has the
+    // correct size.
+    let legacy_smbios_range_start_addr = 0xF0000;
+    let mut legacy_smbios_range_size = 0;
+
     for entry in e820_table {
-        if entry.entry_type() != Some(E820EntryType::RAM) || entry.addr() < min_addr {
+        if entry.addr() == legacy_smbios_range_start_addr
```

Comment:
Optional: while this is more efficient, I find it less readable. The e820 table is small, so I think it is fine to put all the code for finding the range together further down. That would also mean the entry can be found using `e820_table.iter().find(...)` to make the requirements for the range more obvious.

Reply:
Done.